src/snap-confine.apparmor.in

```diff
@@ -85,54 +85,94 @@
     # reading mount profiles
     /{tmp/snap.rootfs_*/,}var/lib/snapd/mount/*.fstab r,
 
-    # set up snap-specific private /tmp dir
-    capability chown,
-    /{tmp/snap.rootfs_*/,}tmp/ w,
-    /{tmp/snap.rootfs_*/,}tmp/snap.*/ w,
-    /{tmp/snap.rootfs_*/,}tmp/snap.*/tmp/ w,
-    mount options=(rw private) ->  /{tmp/snap.rootfs_*/,}tmp/,
-    mount options=(rw bind) /{tmp/snap.rootfs_*/,}tmp/snap.*/tmp/ -> /{tmp/snap.rootfs_*/,}tmp/,
-    mount fstype=devpts options=(rw) devpts -> /{tmp/snap.rootfs_*/,}dev/pts/,
-    mount options=(rw bind) /{tmp/snap.rootfs_*/,}dev/pts/ptmx -> /{tmp/snap.rootfs_*/,}dev/ptmx,     # for bind mounting
-    mount options=(rw bind) /{tmp/snap.rootfs_*/,}dev/pts/ptmx -> /{tmp/snap.rootfs_*/,}dev/pts/ptmx, # for bind mounting under LXD
-    # Workaround for LP: #1584456 on older kernels that mistakenly think
-    # /dev/pts/ptmx needs a trailing '/'
-    mount options=(rw bind) /{tmp/snap.rootfs_*/,}dev/pts/ptmx/ -> /{tmp/snap.rootfs_*/,}dev/ptmx/,
-    mount options=(rw bind) /{tmp/snap.rootfs_*/,}dev/pts/ptmx/ -> /{tmp/snap.rootfs_*/,}dev/pts/ptmx/,
-
-    # for running snaps on classic
-    mount options=(rw rslave) -> /,
-    /tmp/snap.rootfs_*/snap/ r,
-    /tmp/snap.rootfs_*/snap/** r,
-    @SNAP_MOUNT_DIR@/ r,
-    @SNAP_MOUNT_DIR@/** r,
-
-    # mount calls to setup the pivot_root based chroot with the core snap as
-    # the root filesystem.
-    mount options=(rw bind) @SNAP_MOUNT_DIR@/{,ubuntu-}core/*/ -> /tmp/snap.rootfs_*/,
-
+    # boostrapping the mount namespace
+    mount options=(rw rshared) -> /,
+    mount options=(rw bind) /tmp/snap.rootfs_*/ -> /tmp/snap.rootfs_*/,
+    mount options=(rw unbindable) -> /tmp/snap.rootfs_*/,
+    # the next line is for clasic system
+    mount options=(rw rbind) @SNAP_MOUNT_DIR@/{,ubuntu-}core/*/ -> /tmp/snap.rootfs_*/,
+    # the next line is for core system
+    mount options=(rw rbind) / -> /tmp/snap.rootfs_*/,
+    # all of the constructed rootfs is a rslave
+    mount options=(rw rslave) -> /tmp/snap.rootfs_*/,
+    # bidirectional mounts
+    mount options=(rw rbind) /media/ -> /tmp/snap.rootfs_*/media/,
```

Comment:
There's another bidirectional mount when MERGED_USR is set:

  `mount options=(rw rbind) /run/media/ -> /tmp/snap.rootfs_*/run/media/,`

However, I think you said that MERGED_USR is only used on Fedora and we know that SELinux is used there instead of AppArmor so, in practice, this isn't an issue.